src/objc/ModalWebViewController.mm

```diff
@@ -46,6 +47,11 @@ - (void)viewDidLoad {
   // Add the WKWebView to the view hierarchy
   [self.view addSubview:self.webView];
 
+  // Set the custom user agent if provided
+  if (self.customUserAgent) {
```

Comment:
Does this check work with null values or do you need to compare with NULL. Have you tested both cases?